### ISSUE TITLE
versioning and breaking changes

### ISSUE THREAD

**Issue by gregors** [reactjs/react-rails#323]
You guys have done a great job, however, I'd just like to point out that you're not following Semantic Versioning convention which I think most people are assuming you are following.

````
MINOR version when you add functionality in a backwards-compatible manner, ....
````

When you add breaking changes it's polite to change the major version as a signal to everyone that breaking changes have occurred instead of everyone assuming you've added features in a backwards-compatible manner.

I really want to thank you guys for all the work you've done. I'm really enjoying this gem for my present project!

**Comment by rmosolgo** [comment]
Thanks @gregors , did you encounter a specific issue in updating?

**Comment by gregors** [comment]
Yes everything related to your breaking change section, in fact broke my application. Which is to be expected, since they are in fact breaking changes. What wasn't expected is for that to happen in a MINOR version change. To prevent this the MAJOR version should have been incremented, i.e. 2.0.0.

Breaking Changes

    Changed server rendering configuration names #253
    Old 	New
    config.react.timeout 	config.react.server_renderer_timeout
    config.react.max_renderers 	config.react.server_renderer_pool_size
    config.react.react_js 	config.react.server_renderer_options[:files]
    config.react.component_filenames 	config.react.server_renderer_options[:files]
    config.react.replay_console 	config.react.server_renderer_options[:replay_console]
    (none) 	config.react.server_renderer

    JSX is transformed by Babel, not JSTransform #295

**Comment by gregors** [comment]
specifically
ExecJS::ProgramError at /
ReferenceError: [BABEL] unknown: Unknown option: direct.harmony

**Comment by gregors** [comment]
Now when I remote the no longer needed option, everything is fine. I fully understand what is going on, my issue is solely with the expectation of such changes when linked with a MINOR version change. Other than that Happy Friday!

**Comment by lucaspiller** [comment]
I know this is an old thread, but I was just wondering about the same thing as I've got an old version that needs to be updated. If you do switch, please also take into account the semantic versioning of React too.
